### ISSUE TITLE
Add econ to the client

### ISSUE THREAD

**Issue by ChillerDragon** [teeworlds/teeworlds#3149]
Opinions? Concerns?

**Comment by ChillerDragon** [comment]
Should be doable in a 5< line diff commit

![image](https://user-images.githubusercontent.com/20344300/164694974-43680cce-f39a-4adc-b5a5-5410af242397.png)

**Comment by Robyt3** [comment]
Do you mean that the client can connect to servers via econ? Or that the (headless) client can be controlled via econ?

**Comment by heinrich5991** [comment]
Might open issues wrt. cheating. What programs are allowed to connect to that econ socket in normal gameplay?

**Comment by ChillerDragon** [comment]
> Do you mean that the client can connect to servers via econ? Or that the (headless) client can be controlled via econ?

The latter I guess. Adding econ to the client so the client can be controlled using econ. Headless clients would profit from that a lot yea. But also as heinrich said it can be used to upgrade your regular gui client.

> Might open issues wrt. cheating. What programs are allowed to connect to that econ socket in normal gameplay?

If one wants to cheat it is much easier to directly modify the client or use a python script to simulate the keyboard.
The only issue I see is it lowers the barrier for quickly building a chat spam bot.

Not saying that it justifies anything but the ddnet client fifo interface has not yet been used maliciously to my knowledge.

**Comment by heinrich5991** [comment]
> If one wants to cheat it is much easier to directly modify the client or use a python script to simulate the keyboard.

Yes, but that's clearly cheating. Usually, the rule is "everything possible with the original client without modifying inputs is not cheating". Adding econ would create a gray zone.

**Comment by AssassinTee** [comment]
Couldn't you only allow rcon/console to econ clients? I don't see the problem here
